optaplanner-quarkus-integration/optaplanner-quarkus/runtime/src/main/java/org/optaplanner/quarkus/gizmo/GizmoAnnotationWrappers.java

```diff
@@ -0,0 +1,726 @@
+                                            new Object[] {});
+                                    break;
+                                case ARRAY:
+                                default:
+                                    throw new IllegalStateException("Arrays of " + value.componentKind() +
+                                            " are unsupported in the enum parser." +
+                                            " This exception was caused by a" +
+                                            " @" + annotationInstance.name() + " annotation" +
+                                            " for parameter " + value.name() + "." +
+                                            " Maybe put " + annotationInstance.name() +
+                                            " not in a subpackage of org.optaplanner?");
+                            }
+                            break;
+                        case NESTED:
+                            annotationValues.put(value.name(),
+                                    new AnnotationWrapper(value.asNested(), indexView).get());
+                            break;
+                        case UNKNOWN:
+                        default:
+                            throw new IllegalStateException(value.componentKind() +
+                                    " are unsupported in the enum parser." +
+                                    " This exception was caused by a" +
+                                    " @" + annotationInstance.name() + " annotation" +
+                                    " for parameter " + value.name() + "." +
+                                    " Maybe put " + annotationInstance.name() +
+                                    " not in a subpackage of org.optaplanner?");
+                    }
+                });
+            }
+        }
+
+        @Override
+        public Annotation get() {
+            if (annotation != null) {
+                return annotation;
+            } else {
+                return AllOptaPlannerAnnotationEnum.getForClass(annotationType, annotationValues);
+            }
+        }
+
+        public Class<? extends Annotation> getAnnotationType() {
+            return annotationType;
+        }
+
+        public void setAnnotationType(Class<? extends Annotation> annotationType) {
+            this.annotationType = annotationType;
+        }
+
+        public Map<String, Object> getAnnotationValues() {
+            return annotationValues;
+        }
+
+        public void setAnnotationValues(Map<String, Object> annotationValues) {
+            this.annotationValues = annotationValues;
+        }
+
+        public Annotation getAnnotation() {
+            return annotation;
+        }
+
+        public void setAnnotation(Annotation annotation) {
+            this.annotation = annotation;
+        }
+    }
+
+    private static Class<?> findClass(AnnotationInstance annotationInstance, Type type) {
```

Comment:
I suggest these be extracted to separate top-level classes. 700-line long classes that consist pretty much of other classes should IMO be avoided.